packages/blade/src/components/Tag/_decisions/decisions.md

```diff
@@ -0,0 +1,208 @@
+
+<details>
+<summary>Full Controlled Example</summary>
+
+```jsx
+function App() {
+  const [inputValue, setInputValue] = React.useState('');
+  const [tags, setTags] = React.useState([]);
+
+  const addTag = () => {
+    // Add input value to tags and clear the input value
+    setTags([...tags, inputValue]);
+    setInputValue('');
+  }
+
+  const removeTag = (tagName) => {
+    setTags(tags.filter(tagNameValue) => tagNameValue !== tagName);
+  }
+
+  return (
+    <TextInput
+      tagsSlot={tags.map((tagName, index) => (
+        <Tag onDismiss={({ value }) => removeTag(value)}>{tagName}</Tag>
+      ))}
+      value={inputValue}
+      onChange={({ value }) => setInputValue(value)}
+      onKeyDown={(e) => {
+        if (e.key === 'ENTER') {
+          addTag();
+        }
+      }}
+    />
+  );
+}
+```
+
+</details>
+
+<details>
+<summary>Alternate Approach</summary>
+
+#### Alternate Approach: `value` prop
+
+We can extend our `value` prop to accept JSX
+
+```jsx
+<TextInput
+  value={
+    <>
+      <Tag onDismiss={}>kamlesh.chandnani@razorpay.com</Tag>
+      <Tag onDismiss={}>divyanshu.maithani@razopay.com</Tag>
+      saurabhdaw
+    </>
+  }
+  onChange={}
+/>
+```
+
+**Cons**
+
+- Requires us to loop on children and separate out Tags from written word and this will run on every `onChange` event.
+- Have not seen any other library handle it this way
+
+</details>
+
+## Accessibility
+
+- Ensure Close Icon is focussable
+- Close Icon to have appropriate `aria-label`
+- Special keyboard interaction when used inside Input components
+  - Pressing `Tab` should move focus into the input and not Tag
+  - Tag is focussable with Left and Right Arrow Keys
+  - Pressing `Backspace` on Tag should remove it
+  - Clicking on Tag should bring focus on Input
+- Tag should be disabled when used inside a disabled input
+
+Refer to Keyboard Interactions of [react-select](https://react-select.com/home#getting-started) for reference. Other Input components with Tag also follow similar keyboard interactions.
+
+## References
+
+- [react-select](https://react-select.com/home#getting-started)
+- [`TagsInput` component by PluralSight Design System](https://design-system.pluralsight.com/components/tagsinput)
+- [`tokens` prop by Primer AutoComplete.Input](https://primer.style/react/Autocomplete/)
+- [Carbon just putting everything outside of Inputs](https://carbondesignsystem.com/components/tag/usage)
+- [`elemBeforeInput` prop on Atlassian TextField](https://atlassian.design/components/textfield/examples#elements-before-and-after-input)
+
+## Open Questions
+
+- Do we want to discuss Tag's integration with TextInput and TextArea? or should I remove it from API decision altogether?
+  - We can discuss here
+- ~`<Tag />` or `<Tags />` 😅~ `<Tag />`
+- [`tagsSlot` vs `tagSlot` vs `valuePrefix` vs `leading` on Input components](#alternate-prop-name)
+- Should we support `value` prop that allows us to set unique identifier value for Tag?
```

Comment:
since tags are individual components what is the use case of passing the value prop to onDismiss?

Reply:
it can make writing custom handlers a bit simpler in some cases. `onDismiss={dismissHandler}` vs `onDismiss={() => dismissHandler(tagName)}`

Reply:
Not a strong opinion though. This is what atlassian follows

Reply:
hmm, tbh value on non-input thing already makes it difficult to understand. might rather go ahead and do `{() => dismissHandler(tagName)}` because this seems more natural and I can do whatever data I want for this func. 🤔